NitroxClient/MonoBehaviours/Gui/MainMenu/MainMenuMultiplayerPanel.cs

```diff
@@ -145,11 +149,23 @@ private void FindLANServers()
             LANDiscoveryClient.BeginSearching();
             LANDiscoveryClient.ServerFound += (serverEndPoint) =>
             {
-                // mostly copied from LoadSavedServers()
-                CreateServerButton($"{Language.main.Get("Nitrox_ConnectTo")} <b>LAN Server</b>\n{(NitroxPrefs.HideIp.Value ? "****" : serverEndPoint.Address)}:{(NitroxPrefs.HideIp.Value ? "****" : serverEndPoint.Port)}", serverEndPoint.Address.ToString(), serverEndPoint.Port.ToString());
+                // Whatever we insert here is not important because it won't ever be used
+                // It's simply important that have the server registered so that there's no issue when deleting one based on an index
+                ServerList.Instance.Add(new("LAN Server", $"{serverEndPoint.Address}", $"{serverEndPoint.Port}", true));
+                CreateServerButton(MakeButtonText("LAN Server", serverEndPoint.Address, serverEndPoint.Port), $"{serverEndPoint.Address}", $"{serverEndPoint.Port}", true);
             };
         }
 
+        // Two functions meant to regroup the same behaviour
```

Comment:
Comment should not reference code that is below since methods can be reordered in the future by auto formatting. Consider making it an xml doc on the method(s) or removing it.